tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/resources/FontRegistryTest.java

```diff
@@ -41,4 +52,47 @@ public void testBug544026() {
 		assertArrayEquals(fontData, JFaceResources.getDefaultFont().getFontData());
 	}
 
+	@Test
+	public void multipleDisplayDispose() {
+		assumeTrue("multiple Display instance only allowed on Windows", OS.isWindows());
+
+		FontRegistry fontRegistry = new FontRegistry();
+		Display secondDisplay = initializeDisplayInSeparateThread();
+		Font fontOnSecondDisplay = secondDisplay.syncCall(fontRegistry::defaultFont);
+
+		Font fontOnThisDisplayBeforeSecondDisplayDispose = fontRegistry.defaultFont();
+		Device displayOfFontOnSecondDisplay = fontOnSecondDisplay.getDevice();
+		// font registry returns same font for every display
+		assertEquals(secondDisplay, displayOfFontOnSecondDisplay);
+		assertEquals(fontOnThisDisplayBeforeSecondDisplayDispose, fontOnSecondDisplay);
+
+		// after disposing font's display, registry should reinitialize the font
+		secondDisplay.syncExec(secondDisplay::dispose);
+		Font fontOnThisDisplayAfterSecondDisplayDispose = fontRegistry.defaultFont();
+		assertNotEquals(fontOnThisDisplayAfterSecondDisplayDispose, fontOnSecondDisplay);
```

Comment:
I know this should always be the case but I would also check that the font has been disposed too, just in case:

```suggestion
		assertNotEquals(fontOnThisDisplayAfterSecondDisplayDispose, fontOnSecondDisplay);
		
		// When the display is disposed, its fonts are disposed too.
		assertTrue(fontOnSecondDisplay.isDisposed());
```

Reply:
Added in 11c2271.